src/flexmock/api.py

```diff
@@ -311,75 +313,62 @@ def updated(self: Any) -> Any:
+                        )
+                    )
+            elif expected is not raised:
+                raise ExceptionClassError('expected "%s", raised "%s"' % (expected, raised))
 
         def match_return_values(expected: Any, received: Any) -> bool:
-            if not isinstance(expected, tuple):
-                expected = (expected,)
-            if not isinstance(received, tuple):
-                received = (received,)
-            if len(received) != len(expected):
-                return False
-            for i, val in enumerate(received):
-                if not _arguments_match(val, expected[i]):
-                    return False
-            return True
+            expected = (expected,) if not isinstance(expected, tuple) else expected
+            received = (received,) if not isinstance(received, tuple) else received
+            return len(received) == len(expected) and all(
+                _arguments_match(r, e) for r, e in zip(received, expected)
+            )
 
         def pass_thru(
             expectation: Expectation, runtime_self: Any, *kargs: Any, **kwargs: Any
         ) -> Any:
-            return_values = None
             try:
-                original = _getattr(expectation, "_original")
-                _mock = _getattr(expectation, "_mock")
-                if inspect.isclass(_mock):
-                    if type(original) in SPECIAL_METHODS:
-                        original = _getattr(expectation, "_original_function")
-                        return_values = original(*kargs, **kwargs)
-                    else:
-                        return_values = original(runtime_self, *kargs, **kwargs)
-                else:
-                    return_values = original(*kargs, **kwargs)
-            except Exception:
+                return_values = (
+                    (
+                        expectation._original_function(*kargs, **kwargs)
+                        if type(expectation._original) in SPECIAL_METHODS
+                        else expectation._original(runtime_self, *kargs, **kwargs)
+                    )
+                    if inspect.isclass(expectation._mock)
+                    else expectation._original(*kargs, **kwargs)
+                )
+            except Exception:  # pylint: disable=broad-except
                 return _handle_exception_matching(expectation)
-            expected_values = _getattr(expectation, "_return_values")
+            expected_values = expectation._return_values
             if expected_values and not match_return_values(expected_values[0].value, return_values):
-                raise (
-                    MethodSignatureError(
-                        "expected to return %s, returned %s"
-                        % (expected_values[0].value, return_values)
-                    )
+                raise MethodSignatureError(
+                    "expected to return %s, returned %s" % (expected_values[0].value, return_values)
```

Comment:
Notice also added single quotes
```suggestion
                raise MethodSignatureError(f"expected to return '{expected_values[0].value}', returned '{return_values}'")
```